google/cloud/bigquery/_pandas_helpers.py

```diff
@@ -309,10 +308,10 @@ def bq_to_arrow_array(series, bq_field):
         if field_type_upper in schema._STRUCT_TYPES:
             return pyarrow.StructArray.from_pandas(series, type=arrow_type)
         return pyarrow.Array.from_pandas(series, type=arrow_type)
-    except ArrowTypeError:  # pragma: NO COVER
+    except pyarrow.ArrowTypeError:  # pragma: NO COVER
```

Comment:
Do we want to keep the `# pragma: NO COVER` Here?

Reply:
Ideally not. We really should have a test covering this case. I'll see what I can do.

Reply:
Just did a big `# pragma: NO COVER` sweep and was able to remove a ton! One straggler with `shapely` where I've commented the reason why.